tests/test_entities_generic_handler.py

```diff
@@ -0,0 +1,2945 @@
+
+        mocked_sg_project = copy.deepcopy(mock_shotgun.SG_PROJECT)
+        mocked_sg_project[SHOTGUN_JIRA_ID_FIELD] = ""
+
+        mocked_sg_task = copy.deepcopy(mock_shotgun.SG_TASK)
+        mocked_sg_task["project"] = mocked_sg_project
+
+        self.add_to_sg_mock_db(bridge.shotgun, mocked_sg_project)
+        self.add_to_sg_mock_db(bridge.shotgun, mocked_sg_task)
+
+        self.assertFalse(
+            bridge.sync_in_jira(
+                self.HANDLER_NAME,
+                "Task",
+                mock_shotgun.SG_TASK["id"],
+                mock_shotgun.SG_TASK_CHANGE_EVENT,
+            )
+        )
+
+    def test_fptr_to_jira_entity_not_flagged_as_sync(self, mocked_sg):
+        """If the entity in not flagged as synced in FPTR, the event will be rejected."""
+
+        syncer, bridge = self._get_syncer(mocked_sg, name=self.HANDLER_NAME)
+
+        self.add_to_sg_mock_db(bridge.shotgun, mocked_sg.SG_TASK)
+
+        self.assertFalse(
+            bridge.sync_in_jira(
+                self.HANDLER_NAME,
+                "Task",
+                mock_shotgun.SG_TASK["id"],
+                mock_shotgun.SG_TASK_CHANGE_EVENT,
+            )
+        )
+
+    def test_fptr_to_jira_bad_issue_type(self, mocked_sg):
+        """If the FPTR entity is mapped to a bad Jira issue type, the event will be rejected."""
+
+        syncer, bridge = self._get_syncer(mocked_sg, name="entities_generic_bad_jira_issue_type")
+
+        self._mock_sg_data(bridge.shotgun)
+
+        self.assertFalse(
+            bridge.sync_in_jira(
+                "entities_generic_bad_jira_issue_type",
+                "Task",
+                mock_shotgun.SG_TASK["id"],
+                mock_shotgun.SG_TASK_CHANGE_EVENT,
+            )
+        )
+
+    def test_fptr_to_jira_bad_project_jira_key(self, mocked_sg):
+        """
+        If the Jira key associated to the Project the entity we're trying to sync belongs to,
+        doesn't refer to an existing Jira project, the event will be rejected.
+        """
+
+        syncer, bridge = self._get_syncer(mocked_sg, name=self.HANDLER_NAME)
+
+        mocked_sg_project = copy.deepcopy(mock_shotgun.SG_PROJECT)
+        mocked_sg_project[SHOTGUN_JIRA_ID_FIELD] = "Bad Jira Key"
+
+        mocked_sg_task = copy.deepcopy(mock_shotgun.SG_TASK)
+        mocked_sg_task["sg_project"] = mocked_sg_project
+        mocked_sg_task[SHOTGUN_SYNC_IN_JIRA_FIELD] = True
+
+        self.add_to_sg_mock_db(bridge.shotgun, mocked_sg_project)
+        self.add_to_sg_mock_db(bridge.shotgun, mocked_sg_task)
+
+        self.assertFalse(
+            bridge.sync_in_jira(
+                self.HANDLER_NAME,
+                "Task",
+                mock_shotgun.SG_TASK["id"],
+                mock_shotgun.SG_TASK_CHANGE_EVENT,
+            )
+        )
+
+    def test_fptr_to_jira_sync_new_entity(self, mocked_sg):
+        """
+        When a new FPTR entity is synced to Jira, the associated Jira issue will be created.
```

Comment:
This is similar to the both way test but just checking that the default if not specified is to sync both ways/ allow to sync to jira?